projects/cdk/directives/fullscreen/fullscreen.directive.ts

```diff
@@ -0,0 +1,67 @@
+import {DOCUMENT} from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    EventEmitter,
+    inject,
+    Input,
+    Output,
+    signal,
+    ViewChild,
+    ViewEncapsulation,
+} from '@angular/core';
+import {TuiRoot} from '@taiga-ui/core/components/root';
+
+@Component({
+    standalone: true,
+    selector: '[tuiFullscreen]',
+    imports: [TuiRoot],
+    template: '<tui-root #root><ng-content /></tui-root>',
```

Comment:
I think that's not enough to just add nested root, otherwise you will have multiple instances of all dropdowns, portals, dialogs etc. We also need to handle that in here.